### PR TITLE
fix: default protection of sensitive peerID

### PR DESCRIPTION
**Pull request by vickywane**
This PR closes [this](https://github.com/ambianic/ambianic-ui/issues/328)

This PR adds default protection of the peerID shown in the settings page by hiding the peerID by default and providing an icon to toggle the display state.

### CODE REVIEW

**Comment by vickywane** [comment]
@ivelin 

The pending lint error has been fixed in the PR. It had to do with [variable masking](https://docs.worksoft.com/Worksoft_Certify/Certify_Variables/How_To/Masking_Variables.htm).

File: src/views/Timeline.vue [hunk 1]
```diff
@@ -64,15 +62,6 @@
         dense
         class="pa-0 ma-0"
       >
-        <infinite-loading
```
Comment (ivelin) [inline src/views/Timeline.vue:67]: @vickywane Why did you change the semantics of the top and bottom spinner. I don't think you preserved the behavior. Please keep PR scoped to a minimal possible change. This PR should only focus on the PeerID visibility. If you want to suggest change of the top/bottom infinite loader and spinners, please open a separate issue to discuss.

**Review by ivelin — CHANGES_REQUESTED** [review]
Please keep the scope of the PR minimal.

File: src/views/Timeline.vue (hunk 1 above)
Comment (vickywane) [reply src/views/Timeline.vue]: This came in from rebasing. Not an intentional adjustment.

I just restored this file using the copy in the master branch. Is that the latest released version?

**Comment by ivelin** [comment]
@vickywane as I'm reviewing the code, please remember to follow the [semantic release format for commit messages](https://www.conventionalcommits.org/en/v1.0.0/#summary):

`Default protection of sensitive peerID`

should be

`feat: default protection of sensitive peerID; closes#328`

File: cypress/integration/ambianic-tests/remote-connection.spec.js [hunk 2]
```diff
@@ -17,10 +17,13 @@ context('RemoteConnections', () => {
     //     cy.get('#peerID').contains('5568ec87-42d8-47b0-aeea-01a125db0623')
     // })
 
-    it('Should switch to a remote Edge Peer ID', () => {
+    it('Should retrieve and display user PeerID', () => {
         // cy.get('#btn-settings').click()
         cy.get('#remotePeerID').type('917d5f0a-6469-4d33-b5c2-efd858118b74')
         cy.get('#btn-sendRemotePeerID').click()
-        cy.get('#edgePeerID').contains('917d5f0a-6469-4d33-b5c2-efd858118b74')
+
```
Comment (ivelin) [inline cypress/integration/ambianic-tests/remote-connection.spec.js:24]: @vickywane how about an assertion here that the PeerID is NOT visible before its revealed explicitly by clicking the toggle button?

**Comment by vickywane** [comment]
> @vickywane as I'm reviewing the code, please remember to follow the [semantic release format for commit messages](https://www.conventionalcommits.org/en/v1.0.0/#summary):
> 
> `Default protection of sensitive peerID`
> 
> should be
> 
> `feat: default protection of sensitive peerID; closes#328`

My commit message was **feat: added default protection of sensitive peerID**, which contains the **feat** keyword for code commit messages

**Review by ivelin — CHANGES_REQUESTED** [review]
@vickywane its looking good now. Please see the comments for adding an improvement to the test and commit message format.

File: cypress/integration/ambianic-tests/remote-connection.spec.js (hunk 2 above)
Comment (vickywane) [reply cypress/integration/ambianic-tests/remote-connection.spec.js:24]: Okay, I would write a test for it.

**Comment by ivelin** [comment]
> > @vickywane as I'm reviewing the code, please remember to follow the [semantic release format for commit messages](https://www.conventionalcommits.org/en/v1.0.0/#summary):
> > `Default protection of sensitive peerID`
> > should be
> > `feat: default protection of sensitive peerID; closes#328`
> 
> My commit message was **feat: added default protection of sensitive peerID**, which contains the **feat** keyword for code commit messages

Looks like a bug in the bot this time. The checks pass now, but the overall result is still fail from a previous run. I reran in several times and it still doesn't update. Will report the bug to the authors.

Screen Shot 2021-02-02 at 2 19 08 PM

**Comment by vickywane** [comment]
> > > @vickywane as I'm reviewing the code, please remember to follow the [semantic release format for commit messages](https://www.conventionalcommits.org/en/v1.0.0/#summary):
> > > `Default protection of sensitive peerID`
> > > should be
> > > `feat: default protection of sensitive peerID; closes#328`
> > 
> > 
> > My commit message was **feat: added default protection of sensitive peerID**, which contains the **feat** keyword for code commit messages
> 
> Looks like a bug in the bot this time. The checks pass now, but the overall result is still fail from a previous run. I reran in several times and it still doesn't update. Will report the bug to the authors.
> 

The check has never passed in any of my PRs even after rewording and using examples from the docs.

**Comment by ivelin** [comment]
> > > > @vickywane as I'm reviewing the code, please remember to follow the [semantic release format for commit messages](https://www.conventionalcommits.org/en/v1.0.0/#summary):
> > > > `Default protection of sensitive peerID`
> > > > should be
> > > > `feat: default protection of sensitive peerID; closes#328`
> > > 
> > > 
> > > My commit message was **feat: added default protection of sensitive peerID**, which contains the **feat** keyword for code commit messages
> > 
> > 
> > Looks like a bug in the bot this time. The checks pass now, but the overall result is still fail from a previous run. I reran in several times and it still doesn't update. Will report the bug to the authors.
> 
> The check has never passed in any of my PRs even after rewording and using examples from the docs.

Just figured it out. The title of the PR needs to also adhere to the same conventions in order for this check to pass.

**Comment by vickywane** [comment]
Noted

**Comment by vickywane** [comment]
This PR can be left open till the new test suite has been added

**Comment by ivelin** [comment]
> Noted.
> 
> This PR can be left open till the new test suite has been added

Agreed.

**Review by ivelin — APPROVED** [review]
Good to go!
